### ISSUE TITLE
retry_after pushback not going down on recurring incidents when using local API

### ISSUE THREAD

**Issue by rcastellaj** [tdlib/telegram-bot-api#535]
Hi, when using the local API server we'd end up getting a 429 during some moments of burst, and we'd then keep track of the retry_after window and not do anything for that specific chat ID until the time expired. On the next incidence of a 429, after the retry window, if another 429 took place the retry_after window would be higher than the previous time.

This behaviour confused us for a while because recurring incidents (maybe 2-3 days apart per chat ID) were not going down in pushback duration but up. To the point that we ended up getting 12-14 hours of rate limit for certain users.

This report is NOT about 429s being out of place -- in this case they are correctly issued due to our bot's flood. The report is about the fact that the retry_after never went down over time. After getting in touch with bot support and bot talk we were told to try the same directly with api.telegram.org, and that resolved the problem. Upon recurring incidences we have been doing down in retry_after lenghts and our 12-14 hour users are now down to 15 minutes or so, and at this rate by tomorrow they'll be down to the normal 10 second pushback which is very easy to honor for us.

So, in summary:

- When using the local bot API, the retry_after window of 429s never goes down in time but keeps increasing until the maximum values, regardless of the pushback being honored by the client.

- When using api.telegram.org with the same code, and honoring the pushback in the same way, the retry_after in recurring occasions was reduced every time until it reaches very small values.

**Comment by levlam** [comment]
The Bot API server is exactly the same as the server at api.telegram.org. You can increase verbosity level of the server's log to ensure that all `retry_after` values were received from Telegram servers. The same values would have been received from the api.telegram.org server in the same time.
